daemons/dss-scalability-test/app.py

```diff
@@ -0,0 +1,330 @@
+
+
+def upload_bundle(event, context, branch_id):
+    app.log.info("Upload bundle")
+    with tempfile.TemporaryDirectory() as src_dir:
+        # TODO: insert json generator here
+        with tempfile.NamedTemporaryFile(dir=src_dir, suffix=".json", delete=False) as jfh:
+            jfh.write(bytes(generate_sample(), 'UTF-8'))
+            jfh.flush()
+        with tempfile.NamedTemporaryFile(dir=src_dir, suffix=".bin") as fh:
+            fh.write(os.urandom(AWS_MIN_CHUNK_SIZE + 1))
+            fh.flush()
+            start_time = current_time()
+            bundle_output = client.upload(src_dir=src_dir, replica="aws", staging_bucket=test_bucket)
+            return {"bundle_id": bundle_output['bundle_uuid'], "start_time": start_time}
+
+
+def download_bundle(event, context, branch_id):
+    app.log.info("Download bundle")
+    bundle_id = event['bundle']['bundle_id']
+    with tempfile.TemporaryDirectory() as dest_dir:
+        client.download(bundle_id, replica="aws", dest_name=dest_dir)
+    return {}
+
+
+def checkout_bundle(event, context, branch_id):
+    bundle_id = event['bundle']['bundle_id']
+    app.log.info(f"Checkout bundle: {bundle_id}")
+    checkout_output = client.post_bundles_checkout(uuid=bundle_id, replica='aws', email='rkisin@chanzuckerberg.com')
+    return {"job_id": checkout_output['checkout_job_id']}
+
+
+def checkout_status(event, context, branch_id):
+    job_id = event['checkout']['job_id']
+    app.log.info(f"Checkout status job_id: {job_id}")
+    checkout_output = client.get_bundles_checkout(checkout_job_id=job_id)
+    print(str(checkout_output))
+    return {"status": checkout_output['status']}
+
+
+def complete_test(event, context):
+    save_results(event, 'SUCCEEDED')
+
+
+def fallback(event, context, branch_id):
+    return {"failed": "failed"}
+
+
+def save_results(event, result: str):
+    table = dynamodb.Table('scalability_test')
+    start_time = None
+
+    expiration_ttl = int(time.time()) + 14 * 24 * 60 * 60  # 14 days
+
+    fail_count = 0
+    success_count = 0
+    run_id = None
+    execution_id = ''
+    for branch_event in event:
+        if branch_event.get('failed'):
+            fail_count += 1
+        else:
+            success_count += 1
+
+        if run_id is None and "test_run_id" in branch_event:
+            run_id = branch_event["test_run_id"]
+            execution_id = branch_event["execution_id"]
+            start_time = branch_event['bundle']['start_time']
+
+    table.put_item(
+        Item={
+            'run_id': run_id,
+            'execution_id': execution_id,
+            'duration': current_time() - start_time - WAIT_CHECKOUT * 1000,
+            'success_count': success_count,
+            'fail_count': fail_count,
+            'created_on': datetime.datetime.now().isoformat(),
+            'expiration_ttl': expiration_ttl
+        }
+    )
+
+
+@app.sns_topic_subscriber("dss-scalability-test-run-" + os.environ["DSS_DEPLOYMENT_STAGE"])
```

Comment:
In the docs for the daemon, please list the SNS topics, dynamodb table, and their relationship to the state machine.

Reply:
Done